### ISSUE TITLE
StanHeaders.dll: file not recognized: File format not recognized

### ISSUE THREAD

**Issue by aegerton** [stan-dev/rstan#568]
#### Summary:
Compilation error when running 8schools example

#### Reproducible Steps:
I have newly installed rstan on a Windows machine following an update to R version 3.5.1  and attempted to run the 8schools example, but encountered the following compilation error.

#### Current Output:
```
Error in compileCode(f, code, language = language, verbose = verbose) : 
  Compilation ERROR, function(s)/method(s) not created! C:/Users/AAHE/Documents/R/win-library/3.5/StanHeaders/libs/x64/StanHeaders.dll: file not recognized: File format not recognized
collect2.exe: error: ld returned 1 exit status
```

#### RStan Version:
```
rstan_2.17.4
StanHeaders_2.17.2
```
#### R Version:
```
R version 3.5.1 (2018-07-02)
```
#### Operating System:
```
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
```

**Comment by bgoodri** [comment]
Take the -flto stuff out of ~/.R/Makevars

**Comment by aegerton** [comment]
Many thanks for the prompt response, @bgoodri 

Here is the content of the Makevars file:-
```
CXXFLAGS=-O3 -Wno-unused-variable -Wno-unused-function

CXXFLAGS += -Wno-ignored-attributes -Wno-deprecated-declarations

CXX14FLAGS=-O3 -Wno-unused-variable -Wno-unused-function
CXX14 = g++ -std=c++1y
```
With apologies for my ignorance, which -flto stuff, please?

**Comment by bgoodri** [comment]
Ah. If you are still using StanHeaders 2.17 (which is fine), then you don't
need the -std=c++1y in the last line (yet).

**Comment by tobhatt** [comment]
I have the exact same problem. Is there a solution available yet? Would be very much appreciated!

**Comment by bgoodri** [comment]
You could also try reinstalling StanHeaders 2.17 from source.

**Comment by tobhatt** [comment]
I downloaded the version StanHeaders 2.17.2 from the CRAN achieve and installed it. However, the same error still occurs. Do you have another idea?

Thanks a lot for your help!

**Comment by bgoodri** [comment]
And you are using the g++ version 4.9.3 from RTools as opposed to the older
4.6.x version?

**Comment by tobhatt** [comment]
From what I see, I'm using the gcc 4.9.3 version.

**Comment by aegerton** [comment]
@bgoodri I came across the following discussion on the Stan forum:-

https://discourse.mc-stan.org/t/failing-to-install-rstan-2-18-on-win10-x64/6044

I do have both 32bit and 64bit versions on my Windows 7 system.  Based on the advice in that thread, I edited the PATH to remove any references to the 32bit Rtools, added back in the  -std=c++1y into the Makevars file and re-installed the new rstan 2.18.1 and StanHeaders 2.18.0 package versions.

I am delighted to advise the rstan installation went fine and the 8schools example runs perfectly.  Thanks for your support and patience.  It is greatly appreciated.
